src/text-editor.coffee

```diff
@@ -101,8 +101,8 @@ class TextEditor extends Model
     throw new Error("Must pass a config parameter when constructing TextEditors") unless @config?
     throw new Error("Must pass a clipboard parameter when constructing TextEditors") unless @clipboard?
     throw new Error("Must pass a grammarRegistry parameter when constructing TextEditors") unless @grammarRegistry?
-    throw new Error("Must pass an assert parameter when constructing TextEditors") unless @assert?
 
+    @assert ?= @defaultAssert.bind(this)
```

Comment:
This way Atom can continue to use `assert` as it does today and other consumers can provide their own assert if they want.